src/cdk/v2/destinations/tiktok_audience/procWorkflow.yaml

```
@@ -0,0 +1,75 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+    exportAll: true
+  - path: ./config
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+
+steps:
+  - name: validateInput
+    template: |
+      let messageType = .message.type;
+      $.assert(.message.type, "message Type is not present. Aborting message.");
+      $.assert(.message.type.toLowerCase() ==='audiencelist', "Event type " + .message.type.toLowerCase() + " is not supported. Aborting message.");
+      $.assert(.message.properties, "Message properties is not present. Aborting message.");
+      $.assert(.message.properties.listData, "listData is not present inside properties. Aborting message.");
+      $.assert($.containsAll(Object.keys(.message.properties.listData), ["add", "remove"]), "unsupported action type. Aborting message.")
+
+  - name: prepareIdentifiersList
+    description: |
+      Populate list of identifiers to be updated
+    template: |
+      const destinationFields = .message.context.destinationFields.split(", ")
+      const audienceId = .message.context.externalId[0].type.split("-")[1];
+      const isHashRequired = .destination.Config.isHashRequired;
+      const advertiserIds = .metadata.secret.advertiserIds;
+      const hashTraits = function(traits) {
+        const batch_data = [];
+        traits.forEach(function(trait){
+          const arr = [];
+          destinationFields.forEach(function(destinationField){
+            let hashedData = trait[destinationField] ? {
+              "id": isHashRequired ? $.SHA256(trait) : trait,
+              "audience_ids":[audienceId]
+            } :{};
+            arr.push(hashedData); 
+          })
+          batch_data.push(arr);
+        })
+        batch_data
+      };
```

```suggestion
      const hashTraits = function(traits) {
        const batch_data = traits.map(function(trait){
          const arr = destinationFields.map(function(destinationField){
            let hashedData = trait[destinationField] ? {
              "id": isHashRequired ? $.SHA256(trait) : trait,
              "audience_ids":[audienceId]
            } :{};
            return hashedData; 
          })
          return arr;
        })
        batch_data
      };
```
Can we refactor like this ?